experimental/dds/tree2/src/feature-libraries/chunked-forest/codec/chunkEncodingGeneric.ts

```diff
@@ -100,15 +120,37 @@ export function handleShapesAndIdentifiers<TEncodedShape>(
 	};
 }
 
+/**
+ * A tree shape which can can references to it deduplicated using {@link Counter}.
```

Comment:
similar vein, but I wondered why you chose to structure the code like this rather than have shapes support iterables of identifiers/other shapes that they reference and let `handleShapesAndIdentifiers`. I'd guess this is probably more performant for a couple reasons (no intermediate iterator/iterable object, and separating identifiers from shapes would make nodeShape's implementation of this do repeated iteration of its own fields). but that's the only reason I could see. Might be worth documenting that choice here in a privateRemark

Reply:
documented.